src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs

```diff
@@ -66,9 +66,14 @@ public static unsafe int NeedsEscaping(ReadOnlySpan<char> value, JavaScriptEncod
                 return -1;
             }
 
-            fixed (char* ptr = value)
+            // Unfortunately, there is no public API for FindFirstCharacterToEncode(Span<char>) yet,
```

Comment:
What difference does it make if we mark the method itself unsafe or if we use an unsafe block?

Reply:
I take it's just to keep unsafeness scoped to the critical code?

Reply:
Presumably this one can be removed (at least for .NET10) once we introduce a safe API `TextEncoder.FindFirstCharacterToEncode(Span)`.

It's not settled yet, but https://github.com/dotnet/designs/pull/330 wants to change meaning of `unsafe` as method modifier to mean "callers must wrap use unsafe context".